### PR TITLE
Documentation: Adds remark to clarify delete behavior

### PR DESCRIPTION
**Pull request by PramodValavala-MSFT**
## Description

The [`DeleteItemAsync<T>`](https://docs.microsoft.com/en-us/dotnet/api/microsoft.azure.cosmos.container.deleteitemasync?view=azure-dotnet) method never returns the delete item, even when `ItemRequestOptions.EnableContentResponseOnWrite` is set to true. But the return type of the methos is `ItemResponse<T>` which could be confusing to users as pointed out [this issue](https://github.com/Azure/azure-sdk-for-net/issues/26739).

## Type of change

Documentation Update

## Closing issues

resolves Azure/azure-sdk-for-net#26739

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs [hunk 1]
```diff
@@ -827,6 +827,9 @@ public abstract class Container
         /// <param name="requestOptions">(Optional) The options for the item request.</param>
         /// <param name="cancellationToken">(Optional) <see cref="CancellationToken"/> representing request cancellation.</param>
         /// <returns>A <see cref="Task"/> containing a <see cref="ItemResponse{T}"/> which will contain information about the request issued.</returns>
+        /// <remarks>
+        /// The deleted item is never returned, even when <see cref="ItemRequestOptions.EnableContentResponseOnWrite"/> is set to true.
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs:831]: I think maybe simplifying the statement to `The deleted item is never returned` or `<see cref="ItemResponse{T}.Resource"/> is always null. <see href="https://docs.microsoft.com/rest/api/cosmos-db/delete-a-document#body"/>

Comment (PramodValavala-MSFT) [reply Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs:831]: Would this be good?

```
<see cref="ItemResponse{T}.Resource"/> is <see href="https://docs.microsoft.com/rest/api/cosmos-db/delete-a-document#body">always null</see>
```

Comment (ealsur) [reply Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs:831]: Makes sense!

Comment (PramodValavala-MSFT) [reply Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs]: Done! 👍

**Review by ealsur — APPROVED** [review]

**Pull request merged**